lib/matrix_client/session.ex

```diff
@@ -5,7 +5,7 @@ defmodule MatrixClient.Session do
   Starts a new session with a base_url.
   """
   def start_link(url) do
-    Agent.start_link(fn -> %{url: url, rooms: %{}, invites: %{}} end)
+    Agent.start_link(fn -> %{url: url, rooms: %{}, invites: %{}, leaves: %{}} end)
```

Comment:
Why do we need to track _leaves_ in the same way as _rooms_? If a room is left, don't we just remove it form the list?

Reply:
I left it because I thought maybe a user would like it. Changed it to where sync_leave updates the rooms state and got rid of the :leaves key